### PR TITLE
Fix `cmath` to handle some edge cases in complex number calculation

### PR DESCRIPTION
**Pull request by jirheee**
Make  `test_user_object` and `test_abs_overflows` pass for `test_cmath.py`

Related to #1671: Increase CPython test coverage

### CODE REVIEW

**Comment by jirheee** [comment]
I realized I had made a mistake, so I fixed it!

File: vm/src/builtins/complex.rs [hunk 1]
```diff
@@ -228,9 +228,15 @@ impl PyComplex {
     }
 
     #[pymethod(magic)]
-    fn abs(&self) -> f64 {
+    fn abs(&self, vm: &VirtualMachine) -> PyResult<PyArithmeticValue<f64>> {
```
Comment (youknowone) [inline vm/src/builtins/complex.rs:231]:
```suggestion
    fn abs(&self, vm: &VirtualMachine) -> PyResult<f64> {
```
Because `PyArithmeticValue::NotImplemented` is not used, `PyResult<f64>` will be enough.

**Review by youknowone — CHANGES_REQUESTED** [review]

File: Lib/test/test_cmath.py [hunk 2]
```diff
@@ -536,9 +534,6 @@ def test_abs(self):
         self.assertEqual(abs(complex(INF, NAN)), INF)
         self.assertTrue(math.isnan(abs(complex(NAN, NAN))))
 
-    # TODO: RUSTPYTHON
-    @unittest.expectedFailure
-    @requires_IEEE_754
```
Comment (youknowone) [inline Lib/test/test_cmath.py:541]: The upper 2 lines are our temporary change for test, but the last line is not.
Please put `@requires_IEEE_754` back 😄

Comment (jirheee) [reply Lib/test/test_cmath.py:541]: Got it! I restored the `@requires_IEEE_754 ` decorator :)

File: vm/src/builtins/complex.rs (hunk 1 above)
Comment (jirheee) [reply vm/src/builtins/complex.rs]: Oh yeah, you're right. I just fixed and pushed it again 👍

**Review by youknowone — APPROVED** [review]
Thank you for contributing!
